utils/filter_instance.py

```diff
@@ -1,17 +1,21 @@
 import numpy as np
 
+from VrpInstance import VrpInstance
 
-def filter_instance(instance: dict, mask: np.ndarray) -> dict:
+
+def filter_instance(instance: VrpInstance, mask: np.ndarray) -> VrpInstance:
```

Comment:
It might make sense to move this function into `VrpInstance`, as e.g. a `filter(self, mask)` member. We can say something in its docstring that the mask only filters *client* data, so e.g. the vehicles are not touched. I think that's what we want anyway, right? 

AFAICT that'd solve the heterogeneous VRP issue.

The implementation might be a bit longer than what we have here (more like `VrpInstance.replace`), but it's pretty hidden away so I don't think that'll be too problematic.

Reply:
Good idea. Thanks!